container/container.go

```diff
@@ -0,0 +1,297 @@
+ *
+ * Author: Sendu Bala <sb10@sanger.ac.uk>, Ashwini Chhipa <ac55@sanger.ac.uk>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ ******************************************************************************/
+
+package container
+
+// this file has functions for dealing with containers
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+
+	fs "github.com/wtsi-ssg/wr/fs/filepath"
+)
+
+// OperationErr is supplied to Error to define the reasons for the failed
+// operations.
+type OperationErr string
+
+// ErrContainer* are the reasons for the failed operations.
+const (
+	ErrContainerList  OperationErr = "Could not list the containers"
+	ErrContainerStats OperationErr = "Could not get the container stats"
+	ErrContainerKill  OperationErr = "Container could not be killed"
+)
+
+// OperatorErr records an error and a reason that caused it.
+type OperatorErr struct {
+	Type OperationErr // one of our OperationErr constants
+	Err  error
+}
+
+// Error returns an error with the reason for a failed operation.
+func (et OperatorErr) Error() string {
+	msg := string(et.Type)
+	if et.Err != nil {
+		msg += " [" + et.Err.Error() + "]"
+	}
+
+	return msg
+}
+
+// Container struct represents a container type with id, name and client
+// properties.
+type Container struct {
+	ID     string
+	Names  []string
+	client Interactor
+}
+
+// Stats returns the current memory usage (RSS, in MB) and total CPU usage (in
+// seconds) of this container.
+func (c *Container) Stats(ctx context.Context) (*Stats, error) {
+	stats, err := c.client.ContainerStats(ctx, c.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	return stats, nil
+}
+
+// HasName checks if a given name is one of this container's name.
+func (c *Container) HasName(name string) bool {
+	for _, cname := range c.Names {
+		cname = strings.TrimPrefix(cname, "/")
+		if cname == name {
+			return true
+		}
+	}
+
+	return false
+}
+
```

Comment:
Now that we have a few Container methods, it makes sense to have only these in container.go, and move the rest to operator.go (both still in the container package).

Reply:
Moved operator functions to a separate file.